File: Script/Fattoriale_ricorsivo/Fattoriale.c
#include <stdio.h>
#include <stdlib.h>

long fattoriale_rec(int n) {
    if (n == 0 || n == 1) return 1;
    return n * fattoriale_rec(n - 1);
}

int main(int argc, char *argv[]) {
    if (argc != 2) {
        printf("Uso: %s <numero>\n", argv[0]);
        return 1;
    }

    int n = atoi(argv[1]);
    printf("Fattoriale ricorsivo(%d) = %ld\n", n, fattoriale_rec(n));
    return 0;
}

File: Script/Fattoriale_ricorsivo/Fattoriale.rs
use std::env;

fn fattoriale_rec(n: u128) -> u128 {
    if n == 0 || n == 1 {
        1
    } else {
        n * fattoriale_rec(n - 1)
    }
}

fn main() {
    let args: Vec<String> = env::args().collect();
    if args.len() != 2 {
        eprintln!("Uso: {} <numero>", args[0]);
        std::process::exit(1);
    }

    let n: u128 = args[1].parse().expect("Inserisci un numero valido");

    if n > 34 {
        eprintln!("Attenzione: {}! non può essere calcolato con i tipi nativi (overflow oltre 34!)", n);
        std::process::exit(1);
    }

    let risultato = fattoriale_rec(n);
    println!("Fattoriale ricorsivo({}) = {}", n, risultato);
}
